tests/pcap/src/handshake_message.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::*;
+use rtshark::Packet;
+use rtshark::RTSharkBuilder;
+use std::collections::HashSet;
+
+use crate::capture::*;
+
+#[derive(Debug, Clone, Default)]
+pub struct Message {
+    pub message_type: String,
+    pub packet: Packet,
+    pub payloads: Vec<String>,
+    pub frame_num: String,
+}
+
+impl Message {
+    const FRAGMENT: &'static str = "tls.handshake.fragment";
+    const FRAGMENTS_COUNT: &'static str = "tls.handshake.fragment.count";
+
+    pub fn to_bytes(&self) -> Result<Vec<u8>> {
+        let mut bytes = Vec::new();
+        for payload in &self.payloads {
+            let hex = payload.replace(':', "");
+            bytes.extend(&hex::decode(&hex)?[5..]);
+        }
+        Ok(bytes)
+    }
+
+    fn from_packet(packet: Packet, frame_num: String, tcp_payloads: &Vec<Packet>) -> Result<Self> {
+        let message_type = get_metadata(&packet, Builder::MESSAGE_TYPE)
+            .context("Missing handshake message type")?
+            .to_string();
+
+        let mut payload_frames = get_all_metadata(&packet, Self::FRAGMENT);
+        if payload_frames.is_empty() {
+            payload_frames.push(&frame_num);
+        }
+        let payload_frames: HashSet<&str> = HashSet::from_iter(payload_frames);
+
+        let mut payloads = Vec::new();
+        for packet in tcp_payloads {
+            if let Some(frame) = get_metadata(packet, Builder::FRAME_NUM) {
+                if payload_frames.contains(frame) {
+                    let payload = get_metadata(packet, Builder::TCP_PAYLOAD)
+                        .context("Missing tcp payload")?;
+                    payloads.push(payload.to_string());
```

Comment:
what encoding is this payload in?

Reply:
I was going to say "hex", but it's actually "hex-ish". Like "FF:FF:FF:FF" instead of "FFFFFFFF". I changed it to be real hex and added doc comments.